packages/admin/cms-admin/src/form/file/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,175 @@
+                    return fetch(`${apiUrl}/public-upload/files/upload`, {
+                        method: "POST",
+                        body: formData,
+                    });
+                });
+
+                const responses = await Promise.all(fetches)
+                    .then((responses) => Promise.all(responses.map((response) => response.json() as Promise<ApiResponse>)))
+                    .finally(() => {
+                        setUploadingFiles(uploadingFiles.filter((loadingFile) => !acceptedFiles.some((file) => file.name === loadingFile.name)));
+                    });
+
+                const successfulUploadResponses = responses.filter((item) => "id" in item) as SuccessfulApiResponse[];
+                const failedUploads = acceptedFiles.filter((file) => !successfulUploadResponses.some((item) => item.name === file.name));
+
+                failedUploads.forEach((file) => {
+                    setFailedUploads((prevFailedUploads) => [
+                        ...prevFailedUploads,
+                        {
+                            name: file.name,
+                            error: <FormattedMessage id="comet.finalFormFileUpload.uploadFailed" defaultMessage="Upload failed." />,
+                        },
+                    ]);
+                });
+
+                const successfullyUploadedFiles = [...successfulUploadResponses.map(({ id, name, size }) => ({ id, name, size }))];
+
+                if (singleFile) {
+                    if (successfullyUploadedFiles.length) {
+                        input.onChange(successfullyUploadedFiles[0]);
+                    }
+                } else {
+                    input.onChange([...inputValue, ...successfullyUploadedFiles]);
+                }
+            }}
+            onRemove={(fileToRemove) => {
+                setTooManyFilesSelected(false);
+
+                if (singleFile) {
+                    input.onChange(undefined);
+                } else if ("id" in fileToRemove) {
+                    input.onChange(inputValue.filter(({ id }) => id !== fileToRemove.id));
+                }
+
+                if ("error" in fileToRemove) {
+                    setFailedUploads((existingFiles) => existingFiles.filter((failedFile) => failedFile.name !== fileToRemove.name));
+                }
+            }}
+            onDownload={async (file) => {
+                setDownloadingFileIds([...downloadingFileIds, file.id]);
+                fetch(`${apiUrl}/public-upload/files/download/${file.id}`).then(async (response) => {
+                    if (!response.ok) {
+                        setDownloadingFileIds(downloadingFileIds.filter((id) => id !== file.id));
+                        setFailedToDownloadFileIds([...failedToDownloadFileIds, file.id]);
+                        return;
+                    }
+
+                    response.blob().then((blob) => {
+                        saveAs(blob, file.name);
```

Comment:
For DAM files we recently added downloading via URL to prevent having this dependency. Should we add that for Public Uploads as well?

Reply:
That would probably make sense 👍🏼 

Do you know if we need to add some kind of endpoint to enable these URLs or does something already exist?

Reply:
If we do that, we should probably remove the "downloading" indicator, as the downloading progress is then handled by the browser.

Reply:
I'd say "Use the platform" 😁

Reply:
The download url is now called directly instead of using `saveAs()`